daemon/image_delete.go

```diff
@@ -132,15 +146,16 @@ func (daemon *Daemon) imgDeleteHelper(name string, list *[]types.ImageDelete, fi
 				if first {
 					return err
 				}
-
 			}
-
 		}
 	}
 	return nil
 }
 
 func (daemon *Daemon) canDeleteImage(imgID string, force bool) error {
+	directlyUsedBy := []string{}
```

Comment:
Perhaps I misunderstood the word "transitive".
Did you mean if `container.ImageID == imgID`, then it's a transitive link? Forgive me for my stupid question :smile_cat: 

We have to check all the containers to see whether there are containers created from the image. 
Let's see the right side of the picture I posted here https://github.com/docker/docker/pull/13422#issuecomment-113460714 , if we are deleting the tag "ubuntu", and if the container "DB" is hit first, since the `ImageID` of the container `DB` is not equal to the ID of the tag "ubuntu", if we stop the loop and return "errTagRemovable", the tag "ubuntu" will be removed. But in fact it should not be removed because the container "test" is created from the tag "ubuntu".
If we don't divide the two types of links, we have to check all the conditions in the `for _, container := range daemon.List() {` loop, this will make the loop too complicated.